L/legate/build_tarballs.jl

```diff
@@ -0,0 +1,174 @@
+
+# Install Python 3.11 (via miniconda)
+cd ${WORKSPACE}/srcdir
+bash miniconda.sh -b -p ${host_bindir}/miniconda
+
+# Create venv and install configure script dependencies
+${host_bindir}/miniconda/bin/python -m venv ./venv
+source ./venv/bin/activate
+pip install --upgrade pip
+pip install rich typing_extensions packaging
+
+python -c "import rich, typing_extensions, packaging; print('Python deps installed and working!')"
+
+cd ${WORKSPACE}/srcdir/legate
+
+### Set Up CUDA ENV Vars
+
+export CPPFLAGS="${CPPFLAGS} -I${prefix}/include"
+export CFLAGS="${CFLAGS} -I${prefix}/include"
+export LDFLAGS="${LDFLAGS} -L${prefix}/lib -L${prefix}/lib64"
+
+export CUDA_HOME=${prefix}/cuda;
+export PATH=$PATH:$CUDA_HOME/bin
+export CUDACXX=$CUDA_HOME/bin/nvcc
+
+# nvcc thinks the libraries are located inside lib64, but the SDK actually has them in lib
+ln -s ${CUDA_HOME}/lib ${CUDA_HOME}/lib64
+
+export CC="clang"
+export CXX="clang++"
+export BUILD_CXX=$(which clang++)
+export BUILD_CC=$(which clang)
+
+# -- "-DCMAKE_CUDA_HOST_COMPILER=$(which clang++)"
+
+
+./configure \
+    --prefix=${prefix} \
+    --with-cudac=${CUDACXX} \
+    --with-cuda-dir=${CUDA_HOME} \
+    --with-nccl=0 \
+    --with-mpiexec-executable=${bindir}/mpiexec \
+    --with-mpi-dir=${prefix} \
+    --with-zlib-dir=${prefix} \
+    --with-hdf5-vfd-gds=0 \
+    --with-hdf5-dir=${prefix} \
+    --num-threads=${nproc} \
+    --with-cxx=${CXX} \
+    --with-cc=${CC} \
+    --CXXFLAGS="${CPPFLAGS}" \
+    --CFLAGS="${CFLAGS}" \
+    --with-clean \
+    --cmake-executable=${host_bindir}/cmake \
+    -- "-DCMAKE_TOOLCHAIN_FILE=/opt/toolchains/${bb_full_target}/target_${target}_clang.cmake" \
+        "-DCMAKE_CUDA_HOST_COMPILER=$(which clang++)"
+
+
+make install -j ${nproc} PREFIX=${prefix}
+install_license ${WORKSPACE}/srcdir/legate/LICENSE
+"""
+
+augment_platform_block = 
+"""
+using Base.BinaryPlatforms
+module __CUDA
+    $(CUDA.augment)
+end
+
+$(MPI.augment)
+
+function augment_platform!(platform::Platform)
+    augment_mpi!(platform)
+    __CUDA.augment_platform!(platform)
+end
+"""
+
+platforms = CUDA.supported_platforms(; min_version = MIN_CUDA_VERSION, max_version = MAX_CUDA_VERSION)
+platforms = filter(p -> os(p) == "linux", platforms)
+platforms = filter!(p -> arch(p) == "x86_64", platforms) #* SHOULD also support aarch64
+
+#* REMOVE LATER
+# platforms = filter!(p -> VersionNumber(tags(p)["cuda"]) == v"12.8", platforms)
+
+platforms = expand_cxxstring_abis(platforms) # -D_GLIBCXX_USE_CXX11_ABI=0
+platforms = filter!(p -> cxxstring_abi(p) == "cxx11", platforms)
+
+# platforms, mpi_dependencies = MPI.augment_platforms(platforms)
+# filter!(p -> p["mpi"] ∉ ["mpitrampoline", "microsoftmpi"], platforms)
+
+# print(platforms)
+
+# platforms = [platforms[1]]
+
+# also some warnings about avx2 instruction set vs x86_64
+products = [
+    LibraryProduct("liblegate", :liblegate, dont_dlopen = true) # dont_dlopen avoids version `GLIBCXX_3.4.30' not found
```

Comment:
Not dlopening here means that whatever package uses this will have to manually dlopen the library.